design/sync-labels-bmh-to-node.md

```diff
@@ -0,0 +1,374 @@
+# Synchronize Labels between BaremetalHosts and Kubernetes Nodes
+
+There are use cases where certain information is inherently tied to the BMH but
+at the same time, is valuable to the users/operators/schedulers in the K
+workload cluster. For example, a user may wish to place her workloads across
+hosts that are in different failure zones. In Kubernetes, labels on Node objects
+are the primary means by which to solve this problem. The proposal is for CAPM3
+to synchronize a specific set of labels placed on a BMH with labels on the
+corresponding Kubernetes Node running on that BMH.
+
+## User Stories
+
+As a user I want replicas of a specific workload spread out across different
+server racks or geographical locations.
+
+As a user, I would like to place my security sensitive workloads on machines
+that meet specific security requirements. For example, certain hosts may be
+in a fortified zone within a data center or certain hosts may have strong
+hardware based security mechanisms in place (e.g. hardware attestation via TPM).
+
+## Goals
+
+* Implement a label synchronization mechanism between BMH and Node objects
+* Limit the scope of synchronization to only labels matching a configured set
+  of prefixes
+* Support a wide range of label prefixes including those in *.kubernetes.io/*
+
+## Non-Goals
+
+* Synchronize taints or annotation on a BMH with those on the Node
+
+## Proposal
+
+CAPM3 can already map a BMH object to the corresponding Node object. The
+proposal seeks to extend CAPM3 to also perform label synchronization
+between the two.
+
+Synchronization is not just a one time copy from BMH to Node on creation.
+More specifically, synchronization accomplishes the following:
+
+1. Addition/Removal of a label, within the set of predefined prefixes, on the
+  BMH will result in the addition/removal of that label on the corresponding
+  Node. Labels outside the prefix set will be ignored.
```

Comment:
Does prefix here mean "the entire part before the /" or `strings.HasPrefix()`? (I hope the former.)

Reply:
It's the entire prefix part before the /